### ISSUE TITLE
[Enhancement]: Download all files for an audiobook. 

### ISSUE THREAD

**Issue by superpawko** [advplyr/audiobookshelf#580]
### Describe the feature/enhancement

[Enhancement]: Download all files for an audiobook.   Right now I can only download every single file separately.  So for audiobook which have 100 mp3 files, it is difficult to do.

**Comment by Kudjo-Grand** [comment]
You used to be able to download a zip of the entire directory, but that is slightly different than getting all the mp3 files at once. What I do, is convert the separate mp3 files into one, single M4B file which supports chapters and several other nice-to-haves. 

This conversion feature is already part of the "Experimental" settings which can be enabled at the bottom of the "settings" page:
![image](https://user-images.githubusercontent.com/25695143/168494535-2ed027cc-3bc9-4903-9a83-c8782c45320f.png)

Then go into any book's edit menu:
![image](https://user-images.githubusercontent.com/25695143/168494584-f7493732-b2c5-4e88-939f-4e89ec1acf6b.png)

And now you can access the "Manage" tab that has this feature.
![image](https://user-images.githubusercontent.com/25695143/168494672-340729a6-0047-4688-84a4-0af8022d4bf1.png)

I actually just submitted an enhancement suggestion to make this feature even easier to use. https://github.com/advplyr/audiobookshelf/issues/594

What do you think? Would this help so you only ever have to download one file ever? It's helped me a LOT! Hopefully, if can help you too.

**Comment by superpawko** [comment]
Yes  I know about m4b, but it takes too much cpu usage, takes a lot of time and is unnecessary if I only need files. Probably creating archive file is needed.  Why I need this ?  For kids audiobook toys. You have to put audiobooks via cable to the toy.

**Comment by jmgant** [comment]
> it takes too much cpu usage, takes a lot of time and is unnecessary if I only need file

how many titles are you talking about? for a number of reasons, my collection management has benefited greatly by transcoding all mp3s into m4b. it takes a little cpu time, but only once.

**Comment by macr0dev** [comment]
This is definitely a feature that I would like to see come back.  I'd ditch booksonic completely if this was available again.

**Comment by advplyr** [comment]
![image](https://user-images.githubusercontent.com/67830747/230798683-e9473fc2-e906-431f-b2c6-2beb10c3aecc.png)

I made this a download button that will download a zip of the entire item folder. It will only show if the user has download permission.

**Comment by advplyr** [comment]
This ended up being inside a context menu
![image](https://user-images.githubusercontent.com/67830747/232345986-12e32e15-05b1-473b-b5a3-70ee4b403c79.png)

**Comment by advplyr** [comment]
Added in [v2.2.19](https://github.com/advplyr/audiobookshelf/releases/tag/v2.2.19)
